### PR TITLE
support download blobs in rpull

### PR DESCRIPTION
**Pull request by yuchen0cc**

### CODE REVIEW

**Comment by yuchen0cc** [comment]
- add tar header for blobs in userspace convertor

File: cmd/convertor/builder/overlaybd_builder.go [hunk 1]
```diff
@@ -135,6 +131,58 @@ func (e *overlaybdBuilderEngine) Cleanup() {
...
+func (e *overlaybdBuilderEngine) uploadBaseLayer(ctx context.Context) (specs.Descriptor, error) {
+	// add baselayer with tar header
+	tarFile := path.Join(e.workDir, "ext4_64.tar")
+	fdes, err := os.Create(tarFile)
+	if err != nil {
+		return specs.Descriptor{}, errors.Wrapf(err, "failed to create file %s", tarFile)
+	}
+	digester := digest.Canonical.Digester()
+	countWriter := &writeCountWrapper{w: io.MultiWriter(fdes, digester.Hash())}
+	tarWriter := tar.NewWriter(countWriter)
```
Comment (WaberZhuang) [inline cmd/convertor/builder/overlaybd_builder.go:143]: I recommand using the two functions `buildArchiveFromFiles()` and `getFileDesc()` in the `builder_utils.go` file to add tar header for baselayer, which makes the code simpler.

Comment (WaberZhuang) [reply cmd/convertor/builder/overlaybd_builder.go:143]: UPDATE: We found `buildArchiveFromFiles()` may affect reproduciable for convertion, so we decided not to make this change for now. We will deal with this in a later commit.

File: cmd/ctr/rpull.go [hunk 2]
```diff
@@ -98,6 +102,9 @@ func rpull(ctx context.Context, client *containerd.Client, ref string, sn string
 	})
 
 	snLabels := map[string]string{"containerd.io/snapshot/image-ref": ref}
+	if download {
+		snLabels["containerd.io/snapshot/overlaybd/download-remote-blob"] = "download"
```
[the remark this answers is not in the document]
Comment (yuchen0cc) [reply cmd/ctr/rpull.go]: @WaberZhuang OK, done.

**Review by liulanzheng — APPROVED** [review]
LGTM

**Pull request merged**